packages/hooks/src/useCreateOffer.ts

```diff
@@ -144,72 +87,60 @@ export default function useCreateOffer(
       expiredAt: Date | null
       auctionId?: string
     }): Promise<string> => {
-      invariant(signer, ErrorMessages.SIGNER_FALSY)
-      const account = await signer.getAddress()
-
+      setActiveProcess(CreateOfferStep.INITIAL)
       try {
-        // approval if needed
-        if (type === 'SALE') {
-          // fetch asset
-          const { asset } = await sdk.FetchAssetForOffer({ assetId })
-          invariant(asset, ErrorMessages.OFFER_CREATION_FAILED)
-          // creating a new offer of type sale, approval is on the asset
-          await approveCollection({
-            chainId: asset.chainId,
-            collectionAddress: asset.collectionAddress,
-          })
-        } else {
-          // creating a new offer of type buy, approval is on the currency
-          await approveCurrency({
-            currencyId,
-            amount: quantity.mul(unitPrice),
-          })
-        }
+        invariant(signer, ErrorMessages.SIGNER_FALSY)
+        const { asset, currency } = await sdk.FetchAssetForOffer({
+          assetId,
+          currencyId,
+        })
+        invariant(asset, ErrorMessages.OFFER_CREATION_FAILED)
+        invariant(currency, ErrorMessages.OFFER_CREATION_FAILED)
 
-        // fetch offer signature
-        setActiveProcess(CreateOfferStep.SIGNATURE)
         const offer = {
-          type,
-          makerAddress: account.toLowerCase(),
-          assetId: assetId,
-          currencyId: currencyId,
-          quantity: quantity.toString(),
-          unitPrice: unitPrice.toString(),
-          takerAddress: takerAddress?.toLowerCase() || null,
-          auctionId: auctionId || null,
-          expiredAt: expiredAt,
+          chain: asset.chainId,
+          collection: toAddress(asset.collectionAddress),
+          token: asset.tokenId,
+          quantity: quantity,
+          takerAddress: takerAddress ? toAddress(takerAddress) : undefined,
+          expiredAt: expiredAt || undefined,
         }
-        const { createOfferSignature } = await sdk.CreateOfferSignature({
-          offer,
-        })
-        const { eip712Data, timestamp, salt } = createOfferSignature
 
-        // sign data
-        const { domain, types, message /*, primaryType */ } = eip712Data
-        delete types.EIP712Domain // Hack: remove primary type from types to allow ethers detect the main type "Order" (aka: primaryType)
-        const signature = await signer._signTypedData(domain, types, message)
+        if (type === 'SALE')
+          return client.exchange.listToken(
+            {
+              ...offer,
+              unitPrice: {
+                amount: unitPrice,
+                currency: currency.address ? toAddress(currency.address) : null,
```

Comment:
How can the currency be null?
I guess this only the currency's address and the chainId is taken from the offer?
That's weird.

Reply:
Yes agree this one is weird, but I really want to remove the concept of id that we have right now.
We could have "composed ids" like `currency: { chain: ChainId, address?: Address }` or similar for a collection `{ chain: ChainId, address: Address }` and an nft `{ collection: { chain: ChainId, address: Address }, tokenId: string }` or event `{ chain: ChainId, collection: Address, tokenId: string }`
I'm not really happy with that, as it is prone to error with the repetition of the chain/collection.

In that case, the "composed ids" would solve the problem, but then we need to have this consistently, which will be really bad.

Reply:
We could replace the null by an empty address like `0x000...` most protocols are doing something like that. Will keep `null` for now to move things forward